### PR TITLE
Lingering issues for Public Release #86

### PR DESCRIPTION
**Pull request by josiahseaman**
This is our final bug fix round before public release.  It deserves lots of testing.  If you can find time, please test and leave us review comments on GitHub.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
     "gh-pages": "^2.2.0",
-    "jquery": "^3.4.1",
-    "konva": "^4.1.2",
-    "mobx": "^5.15.2",
-    "mobx-react": "^6.1.4",
-    "mobx-state-tree": "^3.15.0",
-    "react": "^16.12.0",
-    "react-dom": "^16.12.0",
+    "jquery": "^3.5.1",
+    "konva": "^4.2.2",
```
Comment (6br) [inline package.json:9]: Just for curiosity's sake, is it possible to update a konva or react-script version to the latest?
I saw some vulnerabilities on react-script dependency.

File: src/PangenomeSchematic.js [hunk 2]
```diff
@@ -61,25 +61,25 @@ class PangenomeSchematic extends React.Component {
...
     console.log("Fetching", filepath);
-    return fetch(process.env.PUBLIC_URL + filepath).then((res) => res.json());
+    return fetch(filepath).then((res) => res.json());
   }
 
   loadJsonCache(url, data) {
     console.log("STEP #6: fetched chunks go into loadJsonCache");
 
-    if (data.json_version !== 16) {
+    if (data.json_version !== 17) {
```
Comment (6br) [inline src/PangenomeSchematic.js:70]: As Simon suggests in #86 , Can we please have some very tiny default data people can browse in the master branch of json_version === 17?

**Review by 6br — COMMENTED** [review]
Thank you for implementation!
I'd like to try this branch, but `yarn start` cannot show anything...
I'd be glad if there is any instruction to visualize sars_covid dataset on this branch.

File: src/PangenomeSchematic.js (hunk 2 above)
Comment (6br) [reply src/PangenomeSchematic.js:70]: current test_data seems outdated.

Comment (josiahseaman) [reply src/PangenomeSchematic.js:70]: v17 is available here: https://drive.google.com/drive/folders/12tyfw5HckE2xJePvgJMY8vjUho84EM9p?usp=sharing

**Comment by 6br** [comment]
I found minor thing when I set row number as 1.

**Comment by 6br** [comment]
I found an awkward behaviour when I click bin link on bin_width != 1.


test23

**Comment by josiahseaman** [comment]
I'm going to start trying to reproduce this error now.  I'm on commit 78bd4159c6de245a0e462e4039aefe91cb13bdc0

**Comment by 6br** [comment]
I'm afraid that the latest commit cannot read any `bin2file.json`. I think this is due to the packages update.

**Comment by josiahseaman** [comment]
Maybe you didn't `npm start` again.  I accidentally did that this morning.

**Comment by josiahseaman** [comment]
It was very difficult to reproduce the bug, but I finally got it by throttling my internet connection.  It appears to be a race condition while navigating at the same time you're updating your bin width zoom level.  I think it's sufficient to just fail silently in this case and trigger a re-render.
image

**Comment by 6br** [comment]
I found the reason. The requested URL was `http://localhost:3000/Schematizetest_data/SARS-CoV-b/bin2file.json`. This is called from `tryJSONpath` function. This function needs to be fixed.

From
```
    function tryJSONpath(event) {
      const url =
        process.env.PUBLIC_URL +
        "test_data/" +
        event.target.value +
        "/bin2file.json";
      if (urlExists(url)) {
        console.log("STEP#1: New Data Source: " + event.target.value);
        self.jsonName = event.target.value;
      }
```

To
```
    function tryJSONpath(event) {
      const url =
        process.env.PUBLIC_URL +
        "/test_data/" +
        event.target.value +
        "/bin2file.json";
      if (urlExists(url)) {
        console.log("STEP#1: New Data Source: " + event.target.value);
        self.jsonName = event.target.value;
      }
```

The prefix "/" is needed on "test_data/"

File: src/ViewportInputsStore.js [hunk 3]
```diff
@@ -158,10 +151,18 @@ RootStore = types
...
     function updateWidth(event) {
-        self.pixelsPerColumn = Math.max(3, Number(event.target.value));
+      self.pixelsPerColumn = checkAndForceMinOrMaxValue(
+        Number(event.target.value),
+        3,
+        30
+      );
     }
 
     function tryJSONpath(event) {
```
Comment (6br) [inline src/ViewportInputsStore.js:168]: L171.   "test_data/" -> "/test_data/"

**Comment by josiahseaman** [comment]
I just pushed a commit with both of those fixes.  See if you can reproduce the link bug.

**Comment by 6br** [comment]
> It was very difficult to reproduce the bug, but I finally got it by throttling my internet connection. It appears to be a race condition while navigating at the same time you're updating your bin width zoom level. I think it's sufficient to just fail silently in this case and trigger a re-render.

Certainly, I agree to leave it as.

**Comment by 6br** [comment]
> I just pushed a commit with both of those fixes. See if you can reproduce the link bug.

These bugs are now solved. Thank you!

**Pull request merged**